api/src/test/java/org/pmiops/workbench/mail/MailServiceImplTest.java

```diff
@@ -101,6 +103,75 @@ public void testSendWelcomeEmail() throws MessagingException, ApiException {
     verify(mandrillApi, times(1)).send(any(MandrillApiKeyAndMessage.class));
   }
 
+  @Test
+  public void testSendWelcomeEmail_throwsMessagingException() throws ApiException {
+    when(msgStatus.getRejectReason()).thenReturn("this was rejected");
+    assertThrows(
+        MessagingException.class,
+        () ->
+            service.sendWelcomeEmail(
+                CONTACT_EMAIL, PASSWORD, FULL_USER_NAME, INSTITUTION_NAME, true, true));
+    verify(mandrillApi, times(1)).send(any());
+  }
+
+  @Test
+  public void testSendWelcomeEmail_throwsApiException() throws MessagingException, ApiException {
+    doThrow(ApiException.class).when(mandrillApi).send(any());
+    assertThrows(
+        MessagingException.class,
+        () ->
+            service.sendWelcomeEmail(
+                CONTACT_EMAIL, PASSWORD, FULL_USER_NAME, INSTITUTION_NAME, true, true));
+    verify(mandrillApi, times(3)).send(any());
+  }
+
+  @Test
+  public void testSendWelcomeEmail_invalidEmail_RTAndCt() throws MessagingException {
```

Comment:
Please add local vars to make it clear what these 6 tests are doing.

Like: `boolean userHasRt = true`

Reply:
Decided to not pursue this at this time. If we would like to pursue this convention in the future, we can discuss it.